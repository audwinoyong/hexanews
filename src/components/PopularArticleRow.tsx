import * as React from 'react'
import Typography from '@material-ui/core/Typography'

type Props = {
  author: string,
  title: string,
}

const PopularArticleRow: React.StatelessComponent<Props> = ({
  author,
  title,
}) => {
  return (
    <div style={{marginTop: 24}}>
      <Typography variant='subheading'>
        {title}
      </Typography>
      <Typography variant='body1' color='textSecondary' style={{ marginTop: 4}}>
        {author}
      </Typography>
    </div>
  )
}

export default PopularArticleRow