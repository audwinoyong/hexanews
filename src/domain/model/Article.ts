/**
 * Model for the article.
 */
export type Article = {
  title: string,
  description: string
  category?: string
  authorId?: string
}