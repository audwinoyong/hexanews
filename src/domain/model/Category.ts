/**
 * Model for category object in our application.
 */

export type Category = {
  _id?: string,
  title: string,
  description: string,
}