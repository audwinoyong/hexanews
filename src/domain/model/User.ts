/**
 * Model for the user.
 */
export type User = {
  _id: string,
  email: string,
  name: string,
  description: string,
  createdAt: string,
}