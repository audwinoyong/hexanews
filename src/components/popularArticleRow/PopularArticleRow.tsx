/**
 * The PopularArticleRow Component for displaying popular article in home container
 */

import * as React from 'react'
import Typography from '@material-ui/core/Typography'
import { Dispatch } from 'redux'
import { push } from 'connected-react-router'

import styles from './styles'
import { Article } from '../../domain/model/Article'

/**
 * All props required by the components
 */
type Props = {
  article: Article,
  authorName: string,
  dispatch: Dispatch<any>,
}

/**
 * Render the PopularArticleRow component
 */
const PopularArticleRow: React.StatelessComponent<Props> = ({
  article,
  authorName,
  dispatch,
}) => {
  return (
    <div style={styles.container} onClick={() => dispatch(push(`articleDetail/${article._id}`))}>
      <Typography
        variant='body1'
        style={{ fontWeight: ' bold' } as any}
      >
        {article.title}
      </Typography>
      <Typography
        variant='body1'
        color='textSecondary'
        style={styles.author}
      >
        {authorName}
      </Typography>
    </div>
  )
}

export default PopularArticleRow