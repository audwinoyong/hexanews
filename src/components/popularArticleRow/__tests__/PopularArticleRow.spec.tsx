/**
 * Unit tests for checking the behavior of PopularArticleRow components
 */

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import PopularArticleRow from '../PopularArticleRow'
import { Article } from '../../../domain/model/Article'

Enzyme.configure({ adapter: new Adapter() })

const mockArticle: Article = {
  _id: '2',
  title: 'def',
  description: 'tester',
  category: 'tester',
  authorId: 'tester',
}

describe('Popular Article Row', () => {
  it('renders Popular Article Row', () => {
    const component = renderer.create(
      <PopularArticleRow
        article={mockArticle}
        authorName='tim'
        dispatch={jest.fn()}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})