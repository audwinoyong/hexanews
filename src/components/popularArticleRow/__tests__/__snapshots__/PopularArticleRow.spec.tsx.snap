// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular Article Row renders Popular Article Row 1`] = `
<div
  className=""
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "marginTop": 24,
    }
  }
>
  <p
    className="MuiTypography-root-98 MuiTypography-body1-107"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    def
  </p>
  <p
    className="MuiTypography-root-98 MuiTypography-body1-107 MuiTypography-colorTextSecondary-121"
    style={
      Object {
        "marginTop": 4,
      }
    }
  >
    tim
  </p>
</div>
`;
